angel_system/berkeley/demo/predictor.py

```diff
@@ -432,31 +433,39 @@ def postprocess_instances(self,
             # Re-order overlapped instances in descending order.
             boxes = boxes[sorted_idxs] if boxes is not None else None
             labels = [labels[k] for k in sorted_idxs] if labels is not None else None
+            
             obj_obj_contact_classes = [obj_obj_contact_classes[k] for k in sorted_idxs] if obj_obj_contact_classes is not None else None
             obj_obj_contact_scores = [obj_obj_contact_scores[k] for k in sorted_idxs] if obj_obj_contact_scores is not None else None
 
             obj_hand_contact_classes = [obj_hand_contact_classes[k] for k in sorted_idxs] if obj_hand_contact_classes is not None else None
             obj_hand_contact_scores = [obj_hand_contact_scores[k] for k in sorted_idxs] if obj_hand_contact_scores is not None else None
 
+            obj_obj_contact_scores = [_.item() for _ in obj_obj_contact_scores] if obj_obj_contact_scores is not None else None
+            obj_hand_contact_scores = [_.item() for _ in obj_hand_contact_scores] if obj_hand_contact_scores is not None else None
+
             masks = [masks[idx] for idx in sorted_idxs] if masks is not None else None
             # assigned_colors = [assigned_colors[idx] for idx in sorted_idxs]
             keypoints = keypoints[sorted_idxs] if keypoints is not None else None
-
-            obj_obj_contact_scores = [_.item() for _ in obj_obj_contact_scores]
-            obj_hand_contact_scores = [_.item() for _ in obj_hand_contact_scores]
+            
+        using_contact = True if obj_obj_contact_classes is not None else False
         
         # post-processing: remove_repeated, remove_multi_states, add_contacts
         if REMOVE_REPEATED_obj:
             boxes, labels, obj_obj_contact_scores, obj_obj_contact_classes, obj_hand_contact_scores, obj_hand_contact_classes = util.remove_repeated_obj(boxes, labels, obj_obj_contact_scores, obj_obj_contact_classes, obj_hand_contact_scores, obj_hand_contact_classes)
         if REMOVE_REPEATED_mutil_states:
             boxes, labels, obj_obj_contact_scores, obj_obj_contact_classes, obj_hand_contact_scores, obj_hand_contact_classes= util.remove_repeated_states(boxes, labels, obj_obj_contact_scores, obj_obj_contact_classes, obj_hand_contact_scores, obj_hand_contact_classes)
 
-        if ADD_CONTACT_STATES:
+        if using_contact:
             contact_flag, Contact_infos, contact_hand_flag, Contact_hand_infos = util.find_contact(boxes, labels, obj_obj_contact_scores, obj_obj_contact_classes, obj_hand_contact_scores, obj_hand_contact_classes)
-
+        else:
+            contact_flag = Contact_infos = contact_hand_flag = Contact_hand_infos = None
         # Determine how many predictions we have left
         num_instances = len(labels)
-        assert num_instances == len(obj_obj_contact_scores) == len(obj_obj_contact_classes) == len(obj_hand_contact_scores) == len(obj_hand_contact_classes)
+        
+        if obj_obj_contact_classes is not None:
```

Comment:
`using_contact` is fed by the same comparison, so could technically use that here instead of duplicating the logic.